open-metadata-implementation/admin-services/admin-services-server/src/main/java/org/odpi/openmetadata/adminservices/OMAGServerAdminServices.java

```diff
@@ -678,10 +678,8 @@ public EventBusConfigResponse getEventBus(String userId, String serverName)
              */
             OMAGServerConfig serverConfig = configStore.getServerConfig(userId, serverName, methodName);
 
-            if (configStore != null)
-            {
-                response.setConfig(serverConfig.getEventBusConfig());
-            }
+            response.setConfig(serverConfig.getEventBusConfig());
+
```

Comment:
I am not happy that this check for null (or any similar null check) is being removed.  The config store connector is pluggable and the logic that wraps it may change.  This check does no harm and ensures that it is resilient to such changes.

I just do not understand why this check is a security vulnerability.  If it is being flagged as such by the scanner, the issue should be losed as a false positive.

Reply:
[Industrial Cybersecurity by Pascal Ackerman](https://learning.oreilly.com/library/view/industrial-cybersecurity/9781788395151/)

Insecure configuration vulnerabilities
Configurations play a key role in the security of an application. Often, systems and applications will run with a default configuration, pulled from the vendor's manual or from the Internet. This makes guessing passwords, bypassing login pages, and finding well-known setup vulnerabilities a breeze. Another form of insecure configuration management is where a configuration is just plain wrong, either from the start or after changes were made that compromise the security of the application or system. This faulty configuration can then end up getting used everywhere in the company.

Common attacks associated with configuration management vulnerabilities are as follows:

Server software flaws or misconfigurations ...

it's from O'Reilly.com and the snippet ends there , but you get the idea.

Reply:
@mandy-chessell it's perhaps worth calling out that the scanners are not claiming that we have these vulnerabilities, it's just that the code "smells" like it could have those type of vulnerabilities. It's a bit like what's considered "best practice"

Reply:
@mandy-chessell Apparently it can be difficult to review the actual SONAR alerts, which would make reviewing PR's like this extremely difficult. Are you able to review the original alerts 

https://sonarcloud.io/project/issues?resolved=false&sonarsourceSecurity=insecure-conf&id=odpi_egeria

if not would it help if I added a break down about why each conditional was being removed ?

Reply:
@mandy-chessell @planetf1 @davidradl 
There are internal requirements that state that,
While you are free to reject these fixes I will however need a commitment that security vulnerabilities will be resolved in a timely manner  ( depending on the severity ) and setting the warning to be a false positive isn't acceptable.
Can I please have confirmation on what the egeria communities policy is with regards to security issues in the egeria code base.

Reply:
Hi Mandy and thanks for your timely response.
There seems to be multiple points raised there , let me start with the easy one.

In every change made under this PR the following is true.
The check for null is performed after the object reference has already been assigned a value in the same module. 
The code that is testing for this branch condition is processing Egeria configuration data.
As the branch condition being tested is impossible the configuration of Egeria is deemed to be at best suspect. 
So to summarise what the tool is reporting is that the Egeria configuration processing is just wrong. 

It's not obvious how any of the changes you mention could possible change what the tool is reporting  and a discussion around a design that allows the injection of null pointer exceptions is out of scope of this PR.

The security security threat is a classification of known type of vulnerability named a "configuration vulnerability" , I first encountered the process as part of "hardening" but I am now seeing it discussed as a topic in it's own right. 
For what it worth , I strongly advise you to adhere to the expected behaviours rather that the the individual instances of a vulnerability , the numbers differences are  logarithmic in nature.

It's not a "code checking" tool , it a "standards adherence " reporting tool , it's the team that's being scanned and your already reporting the results. I pulled this piece of work from the Egeria technical debt backlog, so apologises if the discussion feels a bit like your arguing with a satnav.

Reply:
Indeed I do believe it is a retrograde step to remove checking from the code - but if you want to put it in you can - I have registered my protest and we can fix the resulting NPE if a change is made to the underlying software.  I just dispair that we are making the code less resilient to change in the name of "security".

Most of these tools are designed for a single application - they do not understand about components and frameworks.   They do not understand that there are people with different levels of skill and experience working on the code.  However I would not be without them because they do find genuine problems in the code.  This type of change and reports (such like the one that requests removal of parentheses that the compiler does not need - but help humans understand the logic) are regretable.